app/uk/gov/hmrc/gform/gform/NewFormController.scala

```diff
@@ -443,40 +442,47 @@ class NewFormController(
     ) { implicit request => implicit l => cache => implicit ss => formModelOptics =>
       val queryParams: QueryParams = QueryParams.fromRequest(request)
       for {
-        maybeSubmission <- gformConnector.maybeSubmissionDetails(
-                             FormIdData(cache.retrievals, cache.formTemplate._id, noAccessCode),
-                             cache.form.envelopeId
-                           )
-        res <- maybeSubmission match {
-                 case None =>
-                   Redirect(routes.NewFormController.newOrContinue(formTemplateId).url, queryParams.toPlayQueryParams)
-                     .pure[Future]
-                 case Some(submission)
-                     if submission.submittedDate.isAfter(LocalDateTime.now().minusHours(submittedAgeHours)) =>
-                   acknowledgementPdfService
-                     .getRenderedPdfSize(
-                       cache,
-                       Option.empty[AccessCode],
-                       formModelOptics
-                     )
-                     .map { pdfSize =>
-                       val downloadThenNew: DownloadThenNewFormPage = startNewOrLogout
-                         .bindFromRequest()
-                         .fold(
-                           error => new DownloadThenNewFormPage(cache.formTemplate, error, submission, pdfSize, se),
-                           _ =>
-                             new DownloadThenNewFormPage(
-                               cache.formTemplate,
-                               startNewOrLogout,
-                               submission,
-                               pdfSize,
-                               se
+        maybeSubmission <- gformConnector
+                             .maybeSubmissionDetails(
+                               FormIdData(cache.retrievals, formTemplateId, noAccessCode),
+                               cache.form.envelopeId
                              )
-                         )
-                       Ok(download_then_new(frontendConfig, downloadThenNew))
-                     }
-                 case Some(submission) =>
-                   newForm(formTemplateId, cache.toAuthCacheWithoutForm, QueryParams.fromRequest(request))
+                             .flatMap { maybeCurrentSubmission =>
+                               maybeCurrentSubmission.fold {
+                                 gformConnector.maybeSubmissionDetails(
+                                   FormIdData(cache.retrievals, cache.formTemplateId, noAccessCode),
+                                   cache.form.envelopeId
+                                 )
+                               } { currentSubmission =>
+                                 Some(currentSubmission).pure[Future]
+                               }
+                             }
```

Comment:
We do same logic in `AcknowledgementPdfService.scala` can we hide this into single line invocation?